settings/src/script.js

```diff
@@ -117,6 +117,12 @@ function Main( { userId } ) {
 			// Reset to initial after navigating away from a page.
 			if ( hasEdits ) {
 				edit( record );
+
+				// password is not reset by edit(record) as it's initially set as undefined.
+				// See https://github.com/WordPress/wporg-two-factor/issues/117#issuecomment-1515693367.
+				// we've tried setting its initial value as empty string, but this prevented email from being updated.
+				// See https://github.com/WordPress/wporg-two-factor/issues/175.
+				record.password = undefined;
```

Comment:
Love the removal of the password mutation in `useUser`. I feel like using the `edit` method directly is cleaner, this works for me:

```suggestion
				// password is not reset by edit(record) as it's initially undefined.
				// See https://github.com/WordPress/wporg-two-factor/issues/117#issuecomment-1515693367.
				// we've tried setting its initial value as empty string, but this prevented email from being updated.
				// See https://github.com/WordPress/wporg-two-factor/issues/175.
				edit( {
				    ...record,
				    password: undefined,
				} );
```

Reply:
This also sets `hasEdits: false`, which wasn't happening with `record.password = undefined`

Reply:
This is better, thanks!